core/db/migrate/20130807024301_upgrade_adjustments.rb

```diff
@@ -14,7 +14,8 @@ def up
 
     # Tax adjustments have their sources altered
     Spree::Adjustment.where(:originator_type => "Spree::TaxRate").find_each do |adjustment|
-      adjustment.source = adjustment.originator
+      adjustment.source_id = adjustment.originator_id
+      adjustment.source_type = "Spree::TaxRate"
       adjustment.save
```

Comment:
agreed separate commits would be good. no preference here whether it's on this pr or a different one. possibly easier to include here since you're already changing things.